winter/http/throttling.py

```diff
@@ -0,0 +1,55 @@
+import dataclasses
+from rest_framework.throttling import ScopedRateThrottle
+
+from ..core import annotate
+
+
+@dataclasses.dataclass
+class ThrottlingAnnotation:
+    rate: str
+
+
+def throttling(rate: str):
+    return annotate(ThrottlingAnnotation(rate), single=True)
+
+
+class WinterRateThrottle(ScopedRateThrottle):
+
+    def allow_request(self, request, view):
+
+        rate = self._get_rate(request, view)
+        if rate is None:
+            return True
+
+        self.num_requests, self.duration = self.parse_rate(rate)
+
+        self.key = self.get_cache_key(request, view)
+
+        self.history = self.cache.get(self.key, [])
+        self.now = self.timer()
+
+        while self.history and self.history[-1] <= self.now - self.duration:
+            self.history.pop()
+        if len(self.history) >= self.num_requests:
+            return self.throttle_failure()
+        return self.throttle_success()
+
+    def _get_rate(self, request, view):
```

Comment:
Is there any way to avoid these annotations queries and analysis on each request? Ideally, it should be done once at the configuration stage.

Reply:
I will think about it
I think I will find a way

Reply:
Maybe, create a throttling class for each particular view (if needed)?

Reply:
Done, by creating classes